.github/workflows/node.js-test.yml

```diff
@@ -30,9 +30,24 @@ jobs:
       run: npm test
 
   sizewatcher:
-
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v2
     - run: npx @adobe/sizewatcher
+  semantic-release:
+    runs-on: ubuntu-latest
+    needs: [build]
+    if: ${{  !contains(github.event.head_commit.message, '[ci skip]') && github.ref == 'refs/heads/master' }}
```

Comment:
run only if there is no '[skip ci]' in the message & it is the master branch so this only runs on merging to master

see environment variables for explanation of github.ref or GITHUB_REF as an env variable: https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables